packages/@aws-cdk/aws-ec2/lib/instance-types.ts

```diff
@@ -565,4 +586,21 @@ export class InstanceType {
   public toString(): string {
     return this.instanceTypeIdentifier;
   }
+
+  /**
+   * Returns the CPU architecture for the instance type
+   */
+  private resolveArchitecture(): InstanceArchitecture {
+    const ARM_INSTANCE_CLASSES = new Set<InstanceClass>([
+      InstanceClass.A1, InstanceClass.C6G, InstanceClass.C6GD,
```

Comment:
An alternative to this explicit checking against a set was instead to test the target instanceClass against a regular expression. A regex for testing for Graviton or Graviton2 turns out to be pretty straightforward, e.g.:
```javascript
// matches a1, t4g, c6g, c6gd, c6gn, m6g, m6gd, r6g, r6gd
/^a1|[cmrt][46]g[dn]?$/
```
Thoughts?

Reply:
I think you could target all digits instead of only 4 and 6. The important part is the `g` after the generation.

Reply:
That's a great point. I ended up implementing the regex check and used your feedback.